src/lua/utils.h

```diff
@@ -577,6 +579,93 @@ void luaL_iterator_delete(struct luaL_iterator *it);
 int
 tarantool_lua_utils_init(struct lua_State *L);
 
+/** Same as luaL_checkstring but raises box error. **/
+static inline const char *
+luaT_checkstring(struct lua_State *L, int index)
```

Comment:
The patch is difficult to review. Please split it:
1. Add the new helpers, preferably with unit tests.
2. Switch the code to the new helpers and `ILLEGAL_PARAMS` errors (this is probably best to be done in the patch that updates errors raised by the schema module).
3. Patch the console to check the trace.

Also, let's factor out the trivial patches to a separate PR (patches 1, 2, 4).

Reply:
Moved  `core: build fix for recent gcc` commit to a separate PR #10042.

Reply:
Moved next commits to a separate PR #10043:

- `error: support level with errcode`
- `lua: add box.error.is`